diff --git b/Qt7z/7zip/CPP/7zip/Archive/7z/7zRegister.cpp a/Qt7z/7zip/CPP/7zip/Archive/7z/7zRegister.cpp
index 389b540..0865616 100644
--- b/Qt7z/7zip/CPP/7zip/Archive/7z/7zRegister.cpp
+++ a/Qt7z/7zip/CPP/7zip/Archive/7z/7zRegister.cpp
@@ -18,4 +18,6 @@ REGISTER_ARC_IO_DECREMENT_SIG(
   NArcInfoFlags::kFindSignature,
   NULL);
 
+QT7Z_STATIC_LINK_FORCE_INCLUDE
+
 }}
diff --git b/Qt7z/7zip/CPP/7zip/Common/RegisterArc.h a/Qt7z/7zip/CPP/7zip/Common/RegisterArc.h
index 3421ba1..8b6c757 100644
--- b/Qt7z/7zip/CPP/7zip/Common/RegisterArc.h
+++ a/Qt7z/7zip/CPP/7zip/Common/RegisterArc.h
@@ -75,4 +75,10 @@ void RegisterArc(const CArcInfo *arcInfo) throw();
   struct CRegisterArcDecSig { CRegisterArcDecSig() { sig[0]--; RegisterArc(&g_ArcInfo); }}; \
   static CRegisterArcDecSig g_RegisterArc;
 
+#ifdef QT7Z_STATIC_LINK
+#define QT7Z_STATIC_LINK_FORCE_INCLUDE bool g_forceLink;
+#else
+#define QT7Z_STATIC_LINK_FORCE_INCLUDE
+#endif
+
 #endif
diff --git b/Qt7z/7zip/CPP/7zip/Common/RegisterCodec.h a/Qt7z/7zip/CPP/7zip/Common/RegisterCodec.h
index 7ddb760..d19cf37 100644
--- b/Qt7z/7zip/CPP/7zip/Common/RegisterCodec.h
+++ a/Qt7z/7zip/CPP/7zip/Common/RegisterCodec.h
@@ -104,3 +104,9 @@ void RegisterHasher(const CHasherInfo *hasher) throw();
     static REGISTER_HASHER_NAME(cls) g_RegisterHasher;
 
 #endif
+
+#ifdef QT7Z_STATIC_LINK
+#define QT7Z_STATIC_LINK_FORCE_INCLUDE bool g_forceLink;
+#else
+#define QT7Z_STATIC_LINK_FORCE_INCLUDE
+#endif
diff --git b/Qt7z/7zip/CPP/7zip/Compress/Bcj2Register.cpp a/Qt7z/7zip/CPP/7zip/Compress/Bcj2Register.cpp
index 7a48f91..5873e5c 100644
--- b/Qt7z/7zip/CPP/7zip/Compress/Bcj2Register.cpp
+++ a/Qt7z/7zip/CPP/7zip/Compress/Bcj2Register.cpp
@@ -21,4 +21,6 @@ REGISTER_CODEC_VAR
 
 REGISTER_CODEC(BCJ2)
 
+QT7Z_STATIC_LINK_FORCE_INCLUDE
+
 }}
diff --git b/Qt7z/7zip/CPP/7zip/Compress/BcjRegister.cpp a/Qt7z/7zip/CPP/7zip/Compress/BcjRegister.cpp
index f06dcfe..8dc852b 100644
--- b/Qt7z/7zip/CPP/7zip/Compress/BcjRegister.cpp
+++ a/Qt7z/7zip/CPP/7zip/Compress/BcjRegister.cpp
@@ -14,4 +14,6 @@ REGISTER_FILTER_E(BCJ,
     CCoder(true),
     0x3030103, "BCJ")
 
+QT7Z_STATIC_LINK_FORCE_INCLUDE
+
 }}
diff --git b/Qt7z/7zip/CPP/7zip/Compress/BranchRegister.cpp a/Qt7z/7zip/CPP/7zip/Compress/BranchRegister.cpp
index 6331c1b..52d1bd5 100644
--- b/Qt7z/7zip/CPP/7zip/Compress/BranchRegister.cpp
+++ a/Qt7z/7zip/CPP/7zip/Compress/BranchRegister.cpp
@@ -38,4 +38,6 @@ REGISTER_CODECS_VAR
 
 REGISTER_CODECS(Branch)
 
+QT7Z_STATIC_LINK_FORCE_INCLUDE
+
 }}
diff --git b/Qt7z/7zip/CPP/7zip/Compress/CopyRegister.cpp a/Qt7z/7zip/CPP/7zip/Compress/CopyRegister.cpp
index 7141ab5..bd1e751 100644
--- b/Qt7z/7zip/CPP/7zip/Compress/CopyRegister.cpp
+++ a/Qt7z/7zip/CPP/7zip/Compress/CopyRegister.cpp
@@ -12,4 +12,10 @@ REGISTER_CODEC_CREATE(CreateCodec, CCopyCoder())
 
 REGISTER_CODEC_2(Copy, CreateCodec, CreateCodec, 0, "Copy")
 
+namespace NCopy {
+
+QT7Z_STATIC_LINK_FORCE_INCLUDE
+
+}
+
 }
diff --git b/Qt7z/7zip/CPP/7zip/Compress/Lzma2Register.cpp a/Qt7z/7zip/CPP/7zip/Compress/Lzma2Register.cpp
index 8f279eb..6f09668 100644
--- b/Qt7z/7zip/CPP/7zip/Compress/Lzma2Register.cpp
+++ a/Qt7z/7zip/CPP/7zip/Compress/Lzma2Register.cpp
@@ -19,4 +19,6 @@ REGISTER_CODEC_E(LZMA2,
     0x21,
     "LZMA2")
 
+QT7Z_STATIC_LINK_FORCE_INCLUDE
+
 }}
diff --git b/Qt7z/7zip/CPP/7zip/Compress/LzmaRegister.cpp a/Qt7z/7zip/CPP/7zip/Compress/LzmaRegister.cpp
index c802a99..e665ff7 100644
--- b/Qt7z/7zip/CPP/7zip/Compress/LzmaRegister.cpp
+++ a/Qt7z/7zip/CPP/7zip/Compress/LzmaRegister.cpp
@@ -19,4 +19,6 @@ REGISTER_CODEC_E(LZMA,
     0x30101,
     "LZMA")
 
+QT7Z_STATIC_LINK_FORCE_INCLUDE
+
 }}
diff --git b/Qt7z/7zip/CPP/7zip/Compress/PpmdRegister.cpp a/Qt7z/7zip/CPP/7zip/Compress/PpmdRegister.cpp
index a3ebb5f..a1666c7 100644
--- b/Qt7z/7zip/CPP/7zip/Compress/PpmdRegister.cpp
+++ a/Qt7z/7zip/CPP/7zip/Compress/PpmdRegister.cpp
@@ -19,4 +19,6 @@ REGISTER_CODEC_E(PPMD,
     0x30401,
     "PPMD")
 
+QT7Z_STATIC_LINK_FORCE_INCLUDE
+
 }}
diff --git b/Qt7z/7zip/CPP/7zip/Crypto/7zAesRegister.cpp a/Qt7z/7zip/CPP/7zip/Crypto/7zAesRegister.cpp
index f9d5969..7fe6170 100644
--- b/Qt7z/7zip/CPP/7zip/Crypto/7zAesRegister.cpp
+++ a/Qt7z/7zip/CPP/7zip/Crypto/7zAesRegister.cpp
@@ -14,4 +14,6 @@ REGISTER_FILTER_E(7zAES,
     CEncoder(),
     0x6F10701, "7zAES")
 
+QT7Z_STATIC_LINK_FORCE_INCLUDE
+
 }}
diff --git b/Qt7z/7zip/CPP/7zip/Crypto/MyAesReg.cpp a/Qt7z/7zip/CPP/7zip/Crypto/MyAesReg.cpp
index 2800683..3ffb119 100644
--- b/Qt7z/7zip/CPP/7zip/Crypto/MyAesReg.cpp
+++ a/Qt7z/7zip/CPP/7zip/Crypto/MyAesReg.cpp
@@ -13,4 +13,10 @@ REGISTER_FILTER_E(AES256CBC,
     CAesCbcEncoder(32),
     0x6F00181, "AES256CBC")
 
+namespace NAes {
+
+QT7Z_STATIC_LINK_FORCE_INCLUDE
+
+}
+
 }
diff --git b/Qt7z/7zip/CPP/Common/CRC.cpp a/Qt7z/7zip/CPP/Common/CRC.cpp
index 9a9f81f..3a14c64 100644
--- b/Qt7z/7zip/CPP/Common/CRC.cpp
+++ a/Qt7z/7zip/CPP/Common/CRC.cpp
@@ -5,3 +5,7 @@
 #include "../../C/7zCrc.h"
 
 struct CCRCTableInit { CCRCTableInit() { CrcGenerateTable(); } } g_CRCTableInit;
+
+#ifdef QT7Z_STATIC_LINK
+bool g_forceLinkCRC;
+#endif
